integration/run_test.go

```diff
@@ -157,6 +168,53 @@ func TestRun(t *testing.T) {
 	}
 }
 
+func TestRunTail(t *testing.T) {
+	MarkIntegrationTest(t, CanRunWithoutGcp)
+
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			if test.targetLog == "" {
+				t.SkipNow()
+			}
+			ns, client := SetupNamespace(t)
+
+			args := append(test.args, "--tail")
+			out := skaffold.Run(args...).InDir(test.dir).InNs(ns.Name).WithEnv(test.env).RunOrFailOutput(t)
+
+			client.WaitForPodsReady(test.pods...)
+			client.WaitForDeploymentsToStabilize(test.deployments...)
+
+			rc := bytes.NewReader(out)
+			WaitForLogs(t, rc, test.targetLog)
+
+			skaffold.Delete().InDir(test.dir).InNs(ns.Name).WithEnv(test.env).RunOrFail(t)
+		})
+	}
+}
+
+func TestRunTailDefaultNamespace(t *testing.T) {
+	MarkIntegrationTest(t, CanRunWithoutGcp)
+
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			if test.targetLog == "" {
+				t.SkipNow()
+			}
+			ns, client := DefaultNamespace(t)
+
+			args := append(test.args, "--tail")
+			out := skaffold.Run(args...).InDir(test.dir).WithEnv(test.env).RunLive(t)
+
+			client.WaitForPodsReady(test.pods...)
+			client.WaitForDeploymentsToStabilize(test.deployments...)
+
+			WaitForLogs(t, out, test.targetLog)
+
+			skaffold.Delete().InDir(test.dir).InNs(ns.Name).WithEnv(test.env).RunOrFail(t)
```

Comment:
Shouldn't this do the right thing without the `InNS()`?
```suggestion
			skaffold.Delete().InDir(test.dir).WithEnv(test.env).RunOrFail(t)
```

Reply:
yes, good catch